Client/Frontend/Widgets/SiteTableViewController.swift

```diff
@@ -141,17 +152,14 @@ class SiteTableViewController: UIViewController, UITableViewDelegate, UITableVie
     }
 
     func applyTheme() {
-        navigationController?.navigationBar.barTintColor = UIColor.theme.tableView.headerBackground
-        navigationController?.navigationBar.tintColor = UIColor.theme.general.controlTint
-        navigationController?.navigationBar.titleTextAttributes = [NSAttributedString.Key.foregroundColor: UIColor.theme.tableView.headerTextDark]
+        navigationController?.navigationBar.barTintColor = themeManager.currentTheme.colors.layer1
+        navigationController?.navigationBar.tintColor = themeManager.currentTheme.colors.iconAction
+        navigationController?.navigationBar.titleTextAttributes = [.foregroundColor: themeManager.currentTheme.colors.textPrimary]
         setNeedsStatusBarAppearanceUpdate()
 
-        tableView.backgroundColor = UIColor.theme.homePanel.panelBackground
-        tableView.separatorColor = UIColor.theme.tableView.separator
-        if let rows = tableView.indexPathsForVisibleRows {
-            tableView.reloadRows(at: rows, with: .none)
-            tableView.reloadSections(IndexSet(rows.map { $0.section }), with: .none)
-        }
+        tableView.backgroundColor = themeManager.currentTheme.colors.layer1
+        tableView.separatorColor = themeManager.currentTheme.colors.borderPrimary
+        tableView.reloadData()
```

Comment:
In that case a reloadData is necessary?

Reply:
Yes because the cells update their colors directly in cellForRowAt  and not in the apply theme of the cell itself so even if the cell itself conforms to ThemeApplicable the colors won't update correctly. And some of these cell types are reused in so many places I have no idea if they all share the same color pallet.